core/src/epicli/data/common/ansible/playbooks/roles/preflight/tasks/main.yml

```diff
@@ -11,6 +11,55 @@
     fail_msg: "Your architecture {{ ansible_architecture }} is not supported at {{ ansible_distribution }} by following components in inventory -> {{ unsupported_components_checklist }}."
     success_msg: "Your architecture {{ ansible_architecture }} is supported at {{ ansible_distribution }} by all components in inventory."
 
+- name: Include application vars from applications role
+  include_vars:
+    file: roles/applications/vars/main.yml
+    name: k8s_apps
+  when:
+    - not is_upgrade_run # only for 'apply'
+
+- name: Check if enabled applications are supported by the architecture
+  assert:
+    that: _unsupported_enabled_apps | length == 0
+    fail_msg: "Following application(s) are not supported by {{ _k8s_arch }} and cannot be installed: {{ _unsupported_enabled_apps | map(attribute='name') | join(', ') }}"
+    success_msg: "All enabled applications are supported by {{ _k8s_arch }}"
+  vars:
+    _k8s_arch: >-
+      {{ hostvars[groups['kubernetes_master'][0]]['ansible_architecture'] }}
+    _defined: >-
+      {{ k8s_apps.specification.applications | selectattr('enabled', 'defined') | list }}
+    _undefined: >-
+      {{ k8s_apps.specification.applications | selectattr('enabled', 'undefined') | list }}
+    _enabled_apps: >-
+      {{ (_defined | selectattr('enabled') | list) + _undefined }}
+    _unsupported_enabled_apps: "{{ _enabled_apps | selectattr('name', 'in', unsupported_apps[_k8s_arch]) | list }}"
+  run_once: true
+  delegate_to: localhost
+  when:
+    - not is_upgrade_run # only for apply
+    - groups.kubernetes_master is defined
+    - groups.kubernetes_master | length > 0
+
+- name: Include postgresql vars from postgresql role
+  include_vars:
+    file: roles/postgresql/vars/main.yml
+    name: postgresql
+  when:
+    - not is_upgrade_run # only for 'apply'
+
+- name: Check if PostgreSQL extentions and architecture are not compatible
+  assert:
```

Comment:
Can be changed to 'fail' module, but I decided to keep it the same as other checks